src/main/java/ezschedule/logic/commands/DeleteCommand.java

```diff
@@ -35,27 +35,27 @@ public DeleteCommand(List<Index> targetIndexes) {
     public String commandWord() {
         return COMMAND_WORD;
     }
-
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Event> lastShownList = model.getFilteredEventList();
         StringBuilder feedback = new StringBuilder();
         Collections.sort(targetIndexes); // sort index in order
         Collections.reverse(targetIndexes); // reverse index order to prevent exception when deleting
-        model.clearRecent();
         for (Index targetIndex : targetIndexes) {
             if (targetIndex.getZeroBased() >= lastShownList.size()) {
                 throw new CommandException(
                         String.format(MESSAGE_INVALID_EVENT_DISPLAYED_INDEX, targetIndex.getZeroBased() + 1));
-            } else {
-                model.recentCommands().add(this);
-                Event eventToDelete = lastShownList.get(targetIndex.getZeroBased());
-                model.deleteEvent(eventToDelete);
-                model.addRecentEvent(eventToDelete);
-                feedback.insert(0, eventToDelete.toString());
             }
         }
+        model.clearRecent();
+        for (Index targetIndex: targetIndexes) {
+            model.addRecentCommand(this);
+            Event eventToDelete = lastShownList.get(targetIndex.getZeroBased());
+            model.deleteEvent(eventToDelete);
+            model.addRecentEvent(eventToDelete);
+            feedback.insert(0, eventToDelete.toString());
+        }
         feedback.insert(0, MESSAGE_DELETE_EVENT_SUCCESS);
```

Comment:
why is there a need for double for loop?

Reply:
First loop is to check and ensure that the index provided is valid and doesnt throw any exception.
So in the case which first loop are passed, it suggests that all index provided are valid and proceed on to clearRecent() once and adding all the events to delete into the recentEvent list.